<!DOCTYPE HTML>
<html lang="zh-CN">


<head>
    <meta charset="utf-8">
    <meta name="keywords" content="C++代写：CSCI1300MeasuringDNASimilarity, 留学生CS|辅导|代写">
    <meta name="description" content="代写C++基础作业，分析DNA序列的相似性。  
Homework ProblemThis problem will be comparing and measuring the differences between DNAsequenc">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
    <meta name="renderer" content="webkit|ie-stand|ie-comp">
    <meta name="mobile-web-app-capable" content="yes">
    <meta name="format-detection" content="telephone=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <meta name="referrer" content="no-referrer-when-downgrade">
    <!-- Global site tag (gtag.js) - Google Analytics -->


    <title>C++代写：CSCI1300MeasuringDNASimilarity | 留学生CS|辅导|代写</title>
    <link rel="icon" type="image/png" href="/favicon.png">
    


    <!-- bg-cover style     -->



<link rel="stylesheet" type="text/css" href="/libs/awesome/css/all.min.css">
<link rel="stylesheet" type="text/css" href="/libs/materialize/materialize.min.css">
<link rel="stylesheet" type="text/css" href="/libs/aos/aos.css">
<link rel="stylesheet" type="text/css" href="/libs/animate/animate.min.css">
<link rel="stylesheet" type="text/css" href="/libs/lightGallery/css/lightgallery.min.css">
<link rel="stylesheet" type="text/css" href="/css/matery.css">
<link rel="stylesheet" type="text/css" href="/css/my.css">
<link rel="stylesheet" type="text/css" href="/css/dark.css" media="none" onload="if(media!='all')media='all'">




    <link rel="stylesheet" href="/libs/tocbot/tocbot.css">
    <link rel="stylesheet" href="/css/post.css">




    
        <link rel="stylesheet" type="text/css" href="/css/reward.css">
    



    <script src="/libs/jquery/jquery-3.6.0.min.js"></script>

<meta name="generator" content="Hexo 7.1.1"></head>


<body>
    <header class="navbar-fixed">
    <nav id="headNav" class="bg-color nav-transparent">
        <div id="navContainer" class="nav-wrapper container">
            <div class="brand-logo">
                <a href="/" class="waves-effect waves-light">
                    
                    <img src="/medias/logo.png" class="logo-img" alt="LOGO">
                    
                    <span class="logo-span">留学生CS|辅导|代写</span>
                </a>
            </div>
            

<a href="#" data-target="mobile-nav" class="sidenav-trigger button-collapse"><i class="fas fa-bars"></i></a>
<ul class="right nav-menu">
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/" class="waves-effect waves-light">
      
      <i class="fas fa-home" style="zoom: 0.6;"></i>
      
      <span>首页</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/archives" class="waves-effect waves-light">
      
      <i class="fas fa-archive" style="zoom: 0.6;"></i>
      
      <span>辅导案例集合</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/comment" class="waves-effect waves-light">
      
      <i class="fas fa-comments" style="zoom: 0.6;"></i>
      
      <span>用户评价</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/contact" class="waves-effect waves-light">
      
      <i class="fas fa-user-circle" style="zoom: 0.6;"></i>
      
      <span>联系我</span>
    </a>
    
  </li>
  
  <li>
    <a href="#searchModal" class="modal-trigger waves-effect waves-light">
      <i id="searchIcon" class="fas fa-search" title="搜索" style="zoom: 0.85;"></i>
    </a>
  </li>
  <li>
    <a href="javascript:;" class="waves-effect waves-light" onclick="switchNightMode()" title="深色/浅色模式" >
      <i id="sum-moon-icon" class="fas fa-sun" style="zoom: 0.85;"></i>
    </a>
  </li>
</ul>


<div id="mobile-nav" class="side-nav sidenav">

    <div class="mobile-head bg-color">
        
        <img src="/medias/logo.png" class="logo-img circle responsive-img">
        
        <div class="logo-name">留学生CS|辅导|代写</div>
        <div class="logo-desc">
            
            Never really desperate, only the lost of the soul.
            
        </div>
    </div>

    <ul class="menu-list mobile-menu-list">
        
        <li class="m-nav-item">
	  
		<a href="/" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-home"></i>
			
			首页
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/archives" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-archive"></i>
			
			辅导案例集合
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/comment" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-comments"></i>
			
			用户评价
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/contact" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-user-circle"></i>
			
			联系我
		</a>
          
        </li>
        
        
    </ul>
</div>


        </div>

        
    </nav>

</header>

    



<div class="bg-cover pd-header post-cover" style="background-image: url('/medias/featureimages/8.jpg')">
    <div class="container" style="right: 0px;left: 0px;">
        <div class="row">
            <div class="col s12 m12 l12">
                <div class="brand">
                    <h1 class="description center-align post-title">C++代写：CSCI1300MeasuringDNASimilarity</h1>
                </div>
            </div>
        </div>
    </div>
</div>




<main class="post-container content">

    
    <div class="row">
    <div id="main-content" class="col s12 m12 l9">
        <!-- 文章内容详情 -->
<div id="artDetail">
    <div class="card">
        <div class="card-content article-info">
            <div class="row tag-cate">
                <div class="col s7">
                    
                    <div class="article-tag">
                        
                            <a href="/tags/C-%E4%BB%A3%E5%86%99/">
                                <span class="chip bg-color">C++代写</span>
                            </a>
                        
                    </div>
                    
                </div>
                <div class="col s5 right-align">
                    
                </div>
            </div>

            <div class="post-info">
                
                <div class="post-date info-break-policy">
                    <i class="far fa-calendar-minus fa-fw"></i>发布日期:&nbsp;&nbsp;
                    2019-01-16
                </div>
                

                

                

                

                
            </div>
        </div>
        <hr class="clearfix">

        

        

        <div class="card-content article-card-content">
            <div id="articleContent">
                <p>代写C++基础作业，分析DNA序列的相似性。  </p>
<h2 id="Homework-Problem"><a href="#Homework-Problem" class="headerlink" title="Homework Problem"></a>Homework Problem</h2><p>This problem will be comparing and measuring the differences between DNA<br>sequences. This type of analysis is common in the Computational Biology field.<br>There are three parts to this assignment which will ask you to implement<br>respectively the following functions:</p>
<ul>
<li>similarityScore(…) </li>
<li>countMatches(…) </li>
<li>findBestMatch(…), findBestGenome(…)<br>You are free to choose the names of any other supporting functions. The names<br>should be reflective of the function they perform.<br>In the writeup, when referring to a function we often remove the details of<br>the parameters that the function accepts; i.e. in similarityScore(…) shown<br>above. We do this for brevity. The parameters are described in detail in the<br>problem descriptions listed below.</li>
</ul>
<h2 id="Measuring-DNA-Similarity"><a href="#Measuring-DNA-Similarity" class="headerlink" title="Measuring DNA Similarity"></a>Measuring DNA Similarity</h2><p>DNA is the hereditary material in humans and other species. Almost every cell<br>in a person’s body has the same DNA. All information in a DNA is stored as a<br>code in four chemical bases: adenine (A), guanine (G), cytosine (C) and<br>thymine (T). The differences in the order of these bases provide a means of<br>specifying different information.<br>In the lectures there have been examples of string representation and use in<br>C++. In this assignment we will create strings that represent the possible DNA<br>sequences. DNA sequences use a limited alphabet (A,C,G,T) within a string.<br>Here we will be implementing a number of functions that are used to search for<br>substrings that may represent genes and protein sequences within the DNA.<br>One of the challenges in computational biology is determining where a DNA<br>binding protein will bind in a genome. Each DNA binding protein has a<br>preference for a specific sequence of nucleotides. This preference sequence is<br>known as a motif. The locations of a motif within a genome are important to<br>understanding the behavior of a cell’s response to a changing environment.<br>To find each possible location along the DNA, we must consider how well the<br>motif matches the DNA sequence at each possible position. The protein does not<br>require an exact match to bind and can bind when the sequence is similar to<br>the motif.<br>Another common DNA analysis function is the comparison of newly discovered DNA<br>genomic sequences to the large databases of known DNA sequences and using the<br>similarity of the sequences to help identify the origin of the new sequences.</p>
<h2 id="Hamming-distance-and-similarity-between-two-strings"><a href="#Hamming-distance-and-similarity-between-two-strings" class="headerlink" title="Hamming distance and similarity between two strings"></a>Hamming distance and similarity between two strings</h2><p>Hamming distance is one of the most common ways to measure the similarity<br>between two strings of the same length. Hamming distance is a position-by-<br>position comparison that counts the number of positions in which the<br>corresponding characters in the string are different. Two strings with a small<br>Hamming distance are more similar than two strings with a larger Hamming<br>distance.<br>Example:<br>    first string &#x3D; “ACCT”  second string &#x3D; “ACCG”<br>    A C C T<br>    | | | *<br>    A C C G<br>In this example, there are three matching characters and one mismatch, so the<br>Hamming distance is one.<br>The similarity score for two sequences is then calculated as follows:<br>    similarity_score &#x3D; (string length - hamming distance) &#x2F; string length<br>    similarity_score &#x3D; (4-1)&#x2F;4 &#x3D; 3&#x2F;4 &#x3D; 0.75<br>Two sequences with a high similarity score are more similar than two sequences<br>with a lower similarity score. The two strings are always the same length when<br>calculating a Hamming distance.</p>
<h2 id="Assignment-Details"><a href="#Assignment-Details" class="headerlink" title="Assignment Details"></a>Assignment Details</h2><p>In this assignment, you will calculate the similarity of two sequences using<br>Hamming distance between two strings, search a genomic string looking for<br>matches to a subsequence, and calculate the similarity scores for a sample DNA<br>sequences compared to known DNA sequences.<br>Here we have provided a small portion of a DNA sequence from a human, a mouse,<br>and an unknown species. Smaller DNA sequences will be compared to each of<br>these larger DNA sequences to determine which has the best match. Each of the<br>DNA sequences can be copied from this write-up and stored as a constant or<br>variable in your program.</p>
<h2 id="Part-1-Compare-two-sequences-to-each-other"><a href="#Part-1-Compare-two-sequences-to-each-other" class="headerlink" title="Part 1 - Compare two sequences to each other"></a>Part 1 - Compare two sequences to each other</h2><h3 id="float-similarityScore-string-sequence1-string-sequence2"><a href="#float-similarityScore-string-sequence1-string-sequence2" class="headerlink" title="float similarityScore (string sequence1, string sequence2)"></a>float similarityScore (string sequence1, string sequence2)</h3><p>Define the similarityScore function. This function takes two string arguments,<br>calculates their Hamming distance, and returns their similarity score as a<br>floating-point number. This function should only calculate the similarity if<br>the two strings are the same length, otherwise return 0.<br>You should implement your main function to repeatedly ask to the user for 2<br>strings (sequence1 and sequence2) such that you can pass them as parameters to<br>your similarityScore function. Writing your own main function will allow you<br>to test and debug every function you write.<br>COG will grade Part 1 by generating random sequences to be compared. The<br>results of your function will be compared to our results for the same<br>sequences.</p>
<h2 id="Part-2-Count-the-matches-of-a-sequence-to-a-genome"><a href="#Part-2-Count-the-matches-of-a-sequence-to-a-genome" class="headerlink" title="Part 2 - Count the matches of a sequence to a genome"></a>Part 2 - Count the matches of a sequence to a genome</h2><h3 id="int-countMatches-string-genome-string-sequence1-float-min-score"><a href="#int-countMatches-string-genome-string-sequence1-float-min-score" class="headerlink" title="int countMatches(string genome, string sequence1 , float min_score)"></a>int countMatches(string genome, string sequence1 , float min_score)</h3><p>The countMatches function takes three parameters, a string containing the<br>genome to search, a string containing the sequence to find, and a floating<br>point value containing the minimum similarity score that will be considered a<br>match.<br>The function does the following:</p>
<ul>
<li>The function should find all the positions of the genome where the genome’s substring matches sequence1 with a similarity greater than or equal to the min_score. </li>
<li>The function should return the count of the number of such positions it found.<br>COG will grade Part 2 by generating random sequences to be found in a given<br>genome. The results of your function will be compared to our results for the<br>same sequences.</li>
</ul>
<h2 id="Part-3-Find-best-matching-genome-for-a-given-sequence"><a href="#Part-3-Find-best-matching-genome-for-a-given-sequence" class="headerlink" title="Part 3 - Find best matching genome for a given sequence"></a>Part 3 - Find best matching genome for a given sequence</h2><p>We have a random DNA sequence, and we want to find the closest species to it.<br>Is the DNA sequence more similar to human, mouse, or unknown?<br>When could this kind of comparison be useful? Suppose that the emergency room<br>of some hospital sees a sudden and drastic increase in patients presenting<br>with a particular set of symptoms. Doctors determine the cause to be<br>bacterial, but without knowing the specific species involved they are unable<br>to treat patients effectively. One way of identifying the cause is to obtain a<br>DNA sample and compare it against known bacterial genomes. With a set of<br>similarity scores, doctors can then make more informed decisions regarding<br>treatment, prevention, and tracking of the disease.<br>The goal of this part of the assignment is to write functions that can be<br>useful to determine the identity of different species of bacteria, animals,<br>etc . By simply using the similarity score routine you implemented you can<br>compare an unknown sequence to different genomes and figure out the identity<br>of the unknown sample.</p>
<h3 id="float-findBestMatch-string-genome-string-seq"><a href="#float-findBestMatch-string-genome-string-seq" class="headerlink" title="float findBestMatch(string genome, string seq)"></a>float findBestMatch(string genome, string seq)</h3><p>The findBestMatch function should take two string arguments and return a<br>floating point value of the highest similarity score found for the given<br>sequence at any position within the genome. In other words, this function<br>should traverse the entire genome and find the highest similarity score by<br>using similarityScore() for the comparisons between seq and each sequential<br>substring of genome.<br>hint: this function is very similar in structure to the countMatches function</p>
<h3 id="int-findBestGenome-string-genome1-string-genome2-string-genome3-string"><a href="#int-findBestGenome-string-genome1-string-genome2-string-genome3-string" class="headerlink" title="int findBestGenome(string genome1, string genome2, string genome3, string"></a>int findBestGenome(string genome1, string genome2, string genome3, string</h3><p>seq)</p>
<ul>
<li>The findBestGenome function should take four string arguments(unknown sequence, mouse_genome, human_genome and unknown_genome). </li>
<li>Return an integer indicating which genome string, out of the 3 given, had the highest similarity score with the given sequence. </li>
<li>For each genome, the function will find the highest similarity score of the sequence (at any position) within that genome (call function findBestMatch described above). </li>
<li>The return value from this function will indicate which genome had the best match, 1, 2, or 3. In the case that two or more of the sequences have the same best similarity score, return 0.<br>COG will grade Part 3 based on both the value returned from findBestGenome and<br>findBestMatch.<br>Note: DNA sequences for human, mouse and unknown genomes will be uploaded as a<br>file on Moodle with this assignment for testing purposes.</li>
</ul>
<h2 id="Getting-started"><a href="#Getting-started" class="headerlink" title="Getting started"></a>Getting started</h2><p>Let’s talk about implementation and testing your solutions. We should approach<br>the design of algorithms or programs from the top down. Begin writing your<br>algorithms as very abstract descriptions and then repeatedly writing more<br>detailed descriptions of the complex abstractions. Each of your abstractions<br>could be its own function. Once you have reached a level of detail in your<br>descriptions that provides well-understood solutions to the problem, you can<br>begin to implement (write the code) for those functions. Implementation is<br>best done from the bottom up, meaning you implement the base functions (those<br>functions that don’t call other functions) first and test them until you are<br>satisfied they behave correctly and therefore have the correct code for<br>implementing that function.<br>Once you have your base functions implemented, you can implement the next<br>layer of abstractions that use those base functions in their algorithms.<br>Again, test the new functions until you are satisfied they perform as<br>required. This method of bottom up will progressively add in functionality<br>until the complete algorithm is implemented.<br>This assignment is written in a manner that will allow you to implement<br>functions each part and use COG to verify that it works before implementing<br>the next part.<br>Write your similarityScore() function and test it by calling it from main()<br>and passing it two known strings. For example:<br>    cout &lt;&lt; “Test1: “ &lt;&lt; similarityScore(“ACGT”,”ACGG”) &lt;&lt; endl;<br>    cout &lt;&lt; “Test2: “ &lt;&lt; similarityScore(“ATAT”,”ACAC”) &lt;&lt; endl;<br>    …<br>—|—<br>Once you know that your function works for the specific tests, modify your<br>main to get the input from the user and pass the strings to the function. Once<br>the user input and output are implemented as required, submit to COG to verify<br>your compliance with all the requirements. Now you can implement the code for<br>the next part of the assignment. Repeat the implementation, local testing, and<br>COG testing stages for each of the required functions.<br>When testing the functions, it may be easier to use smaller genome sequences<br>when debugging your code. Use simple examples to verify that the functions are<br>working before testing it on longer strings. Once you’re confident the<br>function works, call your functions from main() using the following strings as<br>the first input parameter to the function:<br>    HumanDNA &#x3D;<br>    “CGCAAATTTGCCGGATTTCCTTTGCTGTTCCTGCATGTAGTTTAAACGAGATTGCC<br>    AGCACCGGGTATCATTCACCATTTTTCTTTTCGTTAACTTGCCGTCAGCCTTTTCTT<br>    TGACCTCTTCTTTCTGTTCATGTGTATTTGCTGTCTCTTAGCCCAGACTTCCCGTGT<br>    CCTTTCCACCGGGCCTTTGAGAGGTCACAGGGTCTTGATGCTGTGGTCTTCATCTG<br>    CAGGTGTCTGACTTCCAGCAACTGCTGGCCTGTGCCAGGGTGCAGCTGAGCACTG<br>    GAGTGGAGTTTTCCTGTGGAGAGGAGCCATGCCTAGAGTGGGATGGGCCATTGTT<br>    CATG”<br>    mouseDNA &#x3D;<br>    “CGCAATTTTTACTTAATTCTTTTTCTTTTAATTCATATATTTTTAATATGTTTACTATT<br>    AATGGTTATCATTCACCATTTAACTATTTGTTATTTTGACGTCATTTTTTTCTATTTCC<br>    TCTTTTTTCAATTCATGTTTATTTTCTGTATTTTTGTTAAGTTTTCACAAGTCTAATAT<br>    AATTGTCCTTTGAGAGGTTATTTGGTCTATATTTTTTTTTCTTCATCTGTATTTTTATG<br>    ATTTCATTTAATTGATTTTCATTGACAGGGTTCTGCTGTGTTCTGGATTGTATTTTTC<br>    TTGTGGAGAGGAACTATTTCTTGAGTGGGATGTACCTTTGTTCTTG”<br>    unknownDNA &#x3D;<br>    “CGCATTTTTGCCGGTTTTCCTTTGCTGTTTATTCATTTATTTTAAACGATATTTATAT<br>    CATCGGGTTTCATTCACTATTTTTCTTTTCGATAAATTTTTGTCAGCATTTTCTTTTAC<br>    CTCTTCTTTCTGTTTATGTTAATTTTCTGTTTCTTAACCCAGTCTTCTCGATTCTTATC<br>    TACCGGACCTATTATAGGTCACAGGGTCTTGATGCTTTGGTTTTCATCTGCAAGAG<br>    TCTGACTTCCTGCTAATGCTGTTCTGTGTCAGGGTGCATCTGAGCACTGATGTGGA<br>    GTTTTCTTGTGGATATGAGCCATTCATAGTGTGGGATGTGCCATAGTTCATG”<br>—|—  </p>
<h2 id="Challenge-Problem-not-graded-by-COG-do-not-submit"><a href="#Challenge-Problem-not-graded-by-COG-do-not-submit" class="headerlink" title="Challenge Problem (not graded by COG, do not submit)"></a>Challenge Problem (not graded by COG, do not submit)</h2><p>DNA is double stranded even though we always write a single strand’s<br>nucleotide sequence. This is because we can infer the other strand’s sequence<br>from the given strand. Every A on one strand has a complementary T on the<br>opposite strand (every C has a G). Therefore we can create the complement<br>strand by swapping the nucleotides with the complementary nucleotide. Create a<br>function to take the sequence and produce the complement sequence.<br>But, you are not done yet. You now have the complement strand, but the DNA is<br>read in the forward direction for the given strand and in the reverse<br>direction for the complementary strand. Therefore, we must reverse the strand<br>(e.g. first character becomes the last character, second becomes second to<br>last, ) to have the correctly specified reverse complement of the given DNA<br>sequence. The challenge problem is to create a function to produce the reverse<br>complement of a given DNA sequence. Once you have a function to produce the<br>reverse complement of a string, you can search both the DNA and reverse<br>complement looking for the best match.</p>

                
            </div>
            <hr/>

            

    <div class="reprint" id="reprint-statement">
        
            <div class="reprint__author">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-user">
                        文章作者:
                    </i>
                </span>
                <span class="reprint-info">
                    <a href="/about" rel="external nofollow noreferrer">SafePoker</a>
                </span>
            </div>
            <div class="reprint__type">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-link">
                        文章链接:
                    </i>
                </span>
                <span class="reprint-info">
                    <a href="http://saferman.github.io/2019/01/16/C++%E4%BB%A3%E5%86%99%EF%BC%9ACSCI1300MeasuringDNASimilarity/">http://saferman.github.io/2019/01/16/C++%E4%BB%A3%E5%86%99%EF%BC%9ACSCI1300MeasuringDNASimilarity/</a>
                </span>
            </div>
            <div class="reprint__notice">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-copyright">
                        版权声明:
                    </i>
                </span>
                <span class="reprint-info">
                    本博客所有文章除特別声明外，均采用
                    <a href="https://creativecommons.org/licenses/by/4.0/deed.zh" rel="external nofollow noreferrer" target="_blank">CC BY 4.0</a>
                    许可协议。转载请注明来源
                    <a href="/about" target="_blank">SafePoker</a>
                    !
                </span>
            </div>
        
    </div>

    <script async defer>
      document.addEventListener("copy", function (e) {
        let toastHTML = '<span>复制成功，请遵循本文的转载规则</span><button class="btn-flat toast-action" onclick="navToReprintStatement()" style="font-size: smaller">查看</a>';
        M.toast({html: toastHTML})
      });

      function navToReprintStatement() {
        $("html, body").animate({scrollTop: $("#reprint-statement").offset().top - 80}, 800);
      }
    </script>



            <div class="tag_share" style="display: block;">
                <div class="post-meta__tag-list" style="display: inline-block;">
                    
                        <div class="article-tag">
                            
                                <a href="/tags/C-%E4%BB%A3%E5%86%99/">
                                    <span class="chip bg-color">C++代写</span>
                                </a>
                            
                        </div>
                    
                </div>
                <div class="post_share" style="zoom: 80%; width: fit-content; display: inline-block; float: right; margin: -0.15rem 0;">
                    <link rel="stylesheet" type="text/css" href="/libs/share/css/share.min.css">
<div id="article-share">

    

    

</div>

                </div>
            </div>
            
                <div id="reward">
    <a href="#rewardModal" class="reward-link modal-trigger btn-floating btn-medium waves-effect waves-light red">赏</a>

    <!-- Modal Structure -->
    <div id="rewardModal" class="modal">
        <div class="modal-content">
            <a class="close modal-close"><i class="fas fa-times"></i></a>
            <h4 class="reward-title">你的赏识是我前进的动力</h4>
            <div class="reward-content">
                <div class="reward-tabs">
                    <ul class="tabs row">
                        <li class="tab col s12 wechat-tab waves-effect waves-light"><a href="#wechat">微 信</a></li>
                    </ul>
                    <div id="wechat">
                        <img src="/medias/newqrcode.jpg" class="reward-img" alt="微信打赏二维码">
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>

<script>
    $(function () {
        $('.tabs').tabs();
    });
</script>

            
        </div>
    </div>

    

    

    

    

    

    

    

    

    

<article id="prenext-posts" class="prev-next articles">
    <div class="row article-row">
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge left-badge text-color">
                <i class="fas fa-chevron-left"></i>&nbsp;上一篇</div>
            <div class="card">
                <a href="/2019/01/17/Python%E4%BB%A3%E5%86%99%EF%BC%9APSY4219GaussiansFilter/">
                    <div class="card-image">
                        
                        
                        <img src="/medias/featureimages/12.jpg" class="responsive-img" alt="Python代写：PSY4219GaussiansFilter">
                        
                        <span class="card-title">Python代写：PSY4219GaussiansFilter</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            
                        
                    </div>
                    <div class="publish-info">
                        <span class="publish-date">
                            <i class="far fa-clock fa-fw icon-date"></i>2019-01-17
                        </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-user fa-fw"></i>
                            SafePoker
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/Python%E4%BB%A3%E5%86%99/">
                        <span class="chip bg-color">Python代写</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge right-badge text-color">
                下一篇&nbsp;<i class="fas fa-chevron-right"></i>
            </div>
            <div class="card">
                <a href="/2019/01/11/Python%E4%BB%A3%E5%86%99%EF%BC%9ACOMP1001MoreTextProcessing/">
                    <div class="card-image">
                        
                        
                        <img src="/medias/featureimages/12.jpg" class="responsive-img" alt="Python代写：COMP1001MoreTextProcessing">
                        
                        <span class="card-title">Python代写：COMP1001MoreTextProcessing</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            
                        
                    </div>
                    <div class="publish-info">
                            <span class="publish-date">
                                <i class="far fa-clock fa-fw icon-date"></i>2019-01-11
                            </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-user fa-fw"></i>
                            SafePoker
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/Python%E4%BB%A3%E5%86%99/">
                        <span class="chip bg-color">Python代写</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
    </div>
</article>

</div>



<!-- 代码块功能依赖 -->
<script type="text/javascript" src="/libs/codeBlock/codeBlockFuction.js"></script>



<!-- 代码语言 -->

<script type="text/javascript" src="/libs/codeBlock/codeLang.js"></script>


<!-- 代码块复制 -->

<script type="text/javascript" src="/libs/codeBlock/codeCopy.js"></script>


<!-- 代码块收缩 -->

<script type="text/javascript" src="/libs/codeBlock/codeShrink.js"></script>



    </div>
    <div id="toc-aside" class="expanded col l3 hide-on-med-and-down">
        <div class="toc-widget card" style="background-color: white;">
            <div class="toc-title"><i class="far fa-list-alt"></i>&nbsp;&nbsp;目录</div>
            <div id="toc-content"></div>
        </div>
    </div>
</div>

<!-- TOC 悬浮按钮. -->

<div id="floating-toc-btn" class="hide-on-med-and-down">
    <a class="btn-floating btn-large bg-color">
        <i class="fas fa-list-ul"></i>
    </a>
</div>


<script src="/libs/tocbot/tocbot.min.js"></script>
<script>
    $(function () {
        tocbot.init({
            tocSelector: '#toc-content',
            contentSelector: '#articleContent',
            headingsOffset: -($(window).height() * 0.4 - 45),
            collapseDepth: Number('0'),
            headingSelector: 'h2, h3, h4'
        });

        // Set scroll toc fixed.
        let tocHeight = parseInt($(window).height() * 0.4 - 64);
        let $tocWidget = $('.toc-widget');
        $(window).scroll(function () {
            let scroll = $(window).scrollTop();
            /* add post toc fixed. */
            if (scroll > tocHeight) {
                $tocWidget.addClass('toc-fixed');
            } else {
                $tocWidget.removeClass('toc-fixed');
            }
        });

        
        /* 修复文章卡片 div 的宽度. */
        let fixPostCardWidth = function (srcId, targetId) {
            let srcDiv = $('#' + srcId);
            if (srcDiv.length === 0) {
                return;
            }

            let w = srcDiv.width();
            if (w >= 450) {
                w = w + 21;
            } else if (w >= 350 && w < 450) {
                w = w + 18;
            } else if (w >= 300 && w < 350) {
                w = w + 16;
            } else {
                w = w + 14;
            }
            $('#' + targetId).width(w);
        };

        // 切换TOC目录展开收缩的相关操作.
        const expandedClass = 'expanded';
        let $tocAside = $('#toc-aside');
        let $mainContent = $('#main-content');
        $('#floating-toc-btn .btn-floating').click(function () {
            if ($tocAside.hasClass(expandedClass)) {
                $tocAside.removeClass(expandedClass).hide();
                $mainContent.removeClass('l9');
            } else {
                $tocAside.addClass(expandedClass).show();
                $mainContent.addClass('l9');
            }
            fixPostCardWidth('artDetail', 'prenext-posts');
        });
        
    });
</script>

    

</main>




    <footer class="page-footer bg-color">
    

    <div class="container row center-align"
         style="margin-bottom: 0px !important;">
        <div class="col s12 m8 l8 copy-right">
            Copyright&nbsp;&copy;
            
                <span id="year">2018-2024</span>
            
            <a href="/contact" target="_blank">SafePoker</a>
            
            <br>
            
            
            
                
            
            
                <span id="busuanzi_container_site_pv">
                &nbsp;|&nbsp;<i class="far fa-eye"></i>&nbsp;总访问量:&nbsp;
                    <span id="busuanzi_value_site_pv" class="white-color"></span>
            </span>
            
            
                <span id="busuanzi_container_site_uv">
                &nbsp;|&nbsp;<i class="fas fa-users"></i>&nbsp;总访问人数:&nbsp;
                    <span id="busuanzi_value_site_uv" class="white-color"></span>
            </span>
            
            <br>

            <!-- 运行天数提醒. -->
            
            <br>
            
        </div>
        <div class="col s12 m4 l4 social-link social-statis">


    <a href="mailto:hhh@gmail.com" class="tooltipped" target="_blank" data-tooltip="邮件联系我" data-position="top" data-delay="50">
        <i class="fas fa-envelope-open"></i>
    </a>













</div>
    </div>
</footer>

<div class="progress-bar"></div>


    <!-- 搜索遮罩框 -->
<div id="searchModal" class="modal">
    <div class="modal-content">
        <div class="search-header">
            <span class="title"><i class="fas fa-search"></i>&nbsp;&nbsp;搜索</span>
            <input type="search" id="searchInput" name="s" placeholder="请输入搜索的关键字"
                   class="search-input">
        </div>
        <div id="searchResult"></div>
    </div>
</div>

<script type="text/javascript">
$(function () {
    var searchFunc = function (path, search_id, content_id) {
        'use strict';
        $.ajax({
            url: path,
            dataType: "xml",
            success: function (xmlResponse) {
                // get the contents from search data
                var datas = $("entry", xmlResponse).map(function () {
                    return {
                        title: $("title", this).text(),
                        content: $("content", this).text(),
                        url: $("url", this).text()
                    };
                }).get();
                var $input = document.getElementById(search_id);
                var $resultContent = document.getElementById(content_id);
                $input.addEventListener('input', function () {
                    var str = '<ul class=\"search-result-list\">';
                    var keywords = this.value.trim().toLowerCase().split(/[\s\-]+/);
                    $resultContent.innerHTML = "";
                    if (this.value.trim().length <= 0) {
                        return;
                    }
                    // perform local searching
                    datas.forEach(function (data) {
                        var isMatch = true;
                        var data_title = data.title.trim().toLowerCase();
                        var data_content = data.content.trim().replace(/<[^>]+>/g, "").toLowerCase();
                        var data_url = data.url;
                        data_url = data_url.indexOf('/') === 0 ? data.url : '/' + data_url;
                        var index_title = -1;
                        var index_content = -1;
                        var first_occur = -1;
                        // only match artiles with not empty titles and contents
                        if (data_title !== '' && data_content !== '') {
                            keywords.forEach(function (keyword, i) {
                                index_title = data_title.indexOf(keyword);
                                index_content = data_content.indexOf(keyword);
                                if (index_title < 0 && index_content < 0) {
                                    isMatch = false;
                                } else {
                                    if (index_content < 0) {
                                        index_content = 0;
                                    }
                                    if (i === 0) {
                                        first_occur = index_content;
                                    }
                                }
                            });
                        }
                        // show search results
                        if (isMatch) {
                            str += "<li><a href='" + data_url + "' class='search-result-title'>" + data_title + "</a>";
                            var content = data.content.trim().replace(/<[^>]+>/g, "");
                            if (first_occur >= 0) {
                                // cut out 100 characters
                                var start = first_occur - 20;
                                var end = first_occur + 80;
                                if (start < 0) {
                                    start = 0;
                                }
                                if (start === 0) {
                                    end = 100;
                                }
                                if (end > content.length) {
                                    end = content.length;
                                }
                                var match_content = content.substr(start, end);
                                // highlight all keywords
                                keywords.forEach(function (keyword) {
                                    var regS = new RegExp(keyword, "gi");
                                    match_content = match_content.replace(regS, "<em class=\"search-keyword\">" + keyword + "</em>");
                                });

                                str += "<p class=\"search-result\">" + match_content + "...</p>"
                            }
                            str += "</li>";
                        }
                    });
                    str += "</ul>";
                    $resultContent.innerHTML = str;
                });
            }
        });
    };

    searchFunc('/search.xml', 'searchInput', 'searchResult');
});
</script>

    <!-- 白天和黑夜主题 -->
<div class="stars-con">
    <div id="stars"></div>
    <div id="stars2"></div>
    <div id="stars3"></div>  
</div>

<script>
    function switchNightMode() {
        $('<div class="Cuteen_DarkSky"><div class="Cuteen_DarkPlanet"></div></div>').appendTo($('body')),
        setTimeout(function () {
            $('body').hasClass('DarkMode') 
            ? ($('body').removeClass('DarkMode'), localStorage.setItem('isDark', '0'), $('#sum-moon-icon').removeClass("fa-sun").addClass('fa-moon')) 
            : ($('body').addClass('DarkMode'), localStorage.setItem('isDark', '1'), $('#sum-moon-icon').addClass("fa-sun").removeClass('fa-moon')),
            
            setTimeout(function () {
            $('.Cuteen_DarkSky').fadeOut(1e3, function () {
                $(this).remove()
            })
            }, 2e3)
        })
    }
</script>

    <!-- 回到顶部按钮 -->
<div id="backTop" class="top-scroll">
    <a class="btn-floating btn-large waves-effect waves-light" href="#!">
        <i class="fas fa-arrow-up"></i>
    </a>
</div>


    <script src="/libs/materialize/materialize.min.js"></script>
    <script src="/libs/masonry/masonry.pkgd.min.js"></script>
    <script src="/libs/aos/aos.js"></script>
    <script src="/libs/scrollprogress/scrollProgress.min.js"></script>
    <script src="/libs/lightGallery/js/lightgallery-all.min.js"></script>
    <script src="/js/matery.js"></script>

    

    
    
    

    <!-- 雪花特效 -->
    

    <!-- 鼠标星星特效 -->
    

     
        <script src="https://ssl.captcha.qq.com/TCaptcha.js"></script>
        <script src="/libs/others/TencentCaptcha.js"></script>
        <button id="TencentCaptcha" data-appid="xxxxxxxxxx" data-cbfn="callback" type="button" hidden></button>
    

    <!-- Baidu Analytics -->

    <!-- Baidu Push -->

<script>
    (function () {
        var bp = document.createElement('script');
        var curProtocol = window.location.protocol.split(':')[0];
        if (curProtocol === 'https') {
            bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';
        } else {
            bp.src = 'http://push.zhanzhang.baidu.com/push.js';
        }
        var s = document.getElementsByTagName("script")[0];
        s.parentNode.insertBefore(bp, s);
    })();
</script>

    
    <script src="/libs/others/clicklove.js" async="async"></script>
    
    
    <script async src="/libs/others/busuanzi.pure.mini.js"></script>
    

    

    

    
    

    

    

    
    <script src="/libs/instantpage/instantpage.js" type="module"></script>
    

</body>

</html>
