<!DOCTYPE HTML>
<html lang="zh-CN">


<head>
    <meta charset="utf-8">
    <meta name="keywords" content="Java代写：CSE142DNA, 留学生CS|辅导|代写">
    <meta name="description" content="熟悉DNA结构，练习基本的Java I&amp;#x2F;O操作，处理DNA数据。  
RequirementThis assignment focuses on arrays and file&amp;#x2F;text processing. Turn">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
    <meta name="renderer" content="webkit|ie-stand|ie-comp">
    <meta name="mobile-web-app-capable" content="yes">
    <meta name="format-detection" content="telephone=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <meta name="referrer" content="no-referrer-when-downgrade">
    <!-- Global site tag (gtag.js) - Google Analytics -->


    <title>Java代写：CSE142DNA | 留学生CS|辅导|代写</title>
    <link rel="icon" type="image/png" href="/favicon.png">
    


    <!-- bg-cover style     -->



<link rel="stylesheet" type="text/css" href="/libs/awesome/css/all.min.css">
<link rel="stylesheet" type="text/css" href="/libs/materialize/materialize.min.css">
<link rel="stylesheet" type="text/css" href="/libs/aos/aos.css">
<link rel="stylesheet" type="text/css" href="/libs/animate/animate.min.css">
<link rel="stylesheet" type="text/css" href="/libs/lightGallery/css/lightgallery.min.css">
<link rel="stylesheet" type="text/css" href="/css/matery.css">
<link rel="stylesheet" type="text/css" href="/css/my.css">
<link rel="stylesheet" type="text/css" href="/css/dark.css" media="none" onload="if(media!='all')media='all'">




    <link rel="stylesheet" href="/libs/tocbot/tocbot.css">
    <link rel="stylesheet" href="/css/post.css">




    
        <link rel="stylesheet" type="text/css" href="/css/reward.css">
    



    <script src="/libs/jquery/jquery-3.6.0.min.js"></script>

<meta name="generator" content="Hexo 7.1.1"></head>


<body>
    <header class="navbar-fixed">
    <nav id="headNav" class="bg-color nav-transparent">
        <div id="navContainer" class="nav-wrapper container">
            <div class="brand-logo">
                <a href="/" class="waves-effect waves-light">
                    
                    <img src="/medias/logo.png" class="logo-img" alt="LOGO">
                    
                    <span class="logo-span">留学生CS|辅导|代写</span>
                </a>
            </div>
            

<a href="#" data-target="mobile-nav" class="sidenav-trigger button-collapse"><i class="fas fa-bars"></i></a>
<ul class="right nav-menu">
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/" class="waves-effect waves-light">
      
      <i class="fas fa-home" style="zoom: 0.6;"></i>
      
      <span>首页</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/archives" class="waves-effect waves-light">
      
      <i class="fas fa-archive" style="zoom: 0.6;"></i>
      
      <span>辅导案例集合</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/comment" class="waves-effect waves-light">
      
      <i class="fas fa-comments" style="zoom: 0.6;"></i>
      
      <span>用户评价</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/contact" class="waves-effect waves-light">
      
      <i class="fas fa-user-circle" style="zoom: 0.6;"></i>
      
      <span>联系我</span>
    </a>
    
  </li>
  
  <li>
    <a href="#searchModal" class="modal-trigger waves-effect waves-light">
      <i id="searchIcon" class="fas fa-search" title="搜索" style="zoom: 0.85;"></i>
    </a>
  </li>
  <li>
    <a href="javascript:;" class="waves-effect waves-light" onclick="switchNightMode()" title="深色/浅色模式" >
      <i id="sum-moon-icon" class="fas fa-sun" style="zoom: 0.85;"></i>
    </a>
  </li>
</ul>


<div id="mobile-nav" class="side-nav sidenav">

    <div class="mobile-head bg-color">
        
        <img src="/medias/logo.png" class="logo-img circle responsive-img">
        
        <div class="logo-name">留学生CS|辅导|代写</div>
        <div class="logo-desc">
            
            Never really desperate, only the lost of the soul.
            
        </div>
    </div>

    <ul class="menu-list mobile-menu-list">
        
        <li class="m-nav-item">
	  
		<a href="/" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-home"></i>
			
			首页
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/archives" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-archive"></i>
			
			辅导案例集合
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/comment" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-comments"></i>
			
			用户评价
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/contact" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-user-circle"></i>
			
			联系我
		</a>
          
        </li>
        
        
    </ul>
</div>


        </div>

        
    </nav>

</header>

    



<div class="bg-cover pd-header post-cover" style="background-image: url('/medias/featureimages/19.jpg')">
    <div class="container" style="right: 0px;left: 0px;">
        <div class="row">
            <div class="col s12 m12 l12">
                <div class="brand">
                    <h1 class="description center-align post-title">Java代写：CSE142DNA</h1>
                </div>
            </div>
        </div>
    </div>
</div>




<main class="post-container content">

    
    <div class="row">
    <div id="main-content" class="col s12 m12 l9">
        <!-- 文章内容详情 -->
<div id="artDetail">
    <div class="card">
        <div class="card-content article-info">
            <div class="row tag-cate">
                <div class="col s7">
                    
                    <div class="article-tag">
                        
                            <a href="/tags/Java%E4%BB%A3%E5%86%99/">
                                <span class="chip bg-color">Java代写</span>
                            </a>
                        
                    </div>
                    
                </div>
                <div class="col s5 right-align">
                    
                </div>
            </div>

            <div class="post-info">
                
                <div class="post-date info-break-policy">
                    <i class="far fa-calendar-minus fa-fw"></i>发布日期:&nbsp;&nbsp;
                    2023-02-12
                </div>
                

                

                

                

                
            </div>
        </div>
        <hr class="clearfix">

        

        

        <div class="card-content article-card-content">
            <div id="articleContent">
                <p>熟悉DNA结构，练习基本的Java I&#x2F;O操作，处理DNA数据。  </p>
<h2 id="Requirement"><a href="#Requirement" class="headerlink" title="Requirement"></a>Requirement</h2><p>This assignment focuses on arrays and file&#x2F;text processing. Turn in a file<br>named DNA.java. You will also need the two input files dna.txt and ecoli.txt<br>from the course web site. Save these files in the same folder as your program.<br>The assignment involves processing data from genome files. Your program should<br>work with the two given input files. If you are curious (this is not<br>required), the National Center for Biotechnology Information publishes many<br>other bacteria genome files. The last page tells you how to use your program<br>to process other published genome files.</p>
<h2 id="Background-Information-About-DNA"><a href="#Background-Information-About-DNA" class="headerlink" title="Background Information About DNA"></a>Background Information About DNA</h2><p>Note: This section explains some information from the field of biology that is<br>related to this assignment. It is for your information only; you do not need<br>to fully understand it to complete the assignment.<br>Deoxyribonucleic acid (DNA) is a complex biochemical macromolecule that<br>carries genetic information for cellular life forms and some viruses. DNA is<br>also the mechanism through which genetic information from parents is passed on<br>during reproduction. DNA consists of long chains of chemical compounds called<br>nucleotides. Four nucleotides are present in DNA: Adenine (A), Cytosine (C),<br>Guanine (G), and Thymine (T). DNA has a double-helix structure (see diagram<br>below) containing complementary chains of these four nucleotides connected by<br>hydrogen bonds.<br>Certain regions of the DNA are called genes. Most genes encode instructions<br>for building proteins (they’re called “protein-coding” genes). These proteins<br>are responsible for carrying out most of the life processes of the organism.<br>Nucleotides in a gene are organized into codons. Codons are groups of three<br>nucleotides and are written as the first letters of their nucleotides (e.g.,<br>TAC or GGA). Each codon uniquely encodes a single amino acid, a building block<br>of proteins.<br>The process of building proteins from DNA has two major phases called<br>transcription and translation, in which a gene is replicated into an<br>intermediate form called mRNA, which is then processed by a structure called a<br>ribosome to build the chain of amino acids encoded by the codons of the gene.<br>The sequences of DNA that encode proteins occur between a start codon (which<br>we will assume to be ATG) and a stop codon (which is any of TAA, TAG, or TGA).<br>Not all regions of DNA are genes; large portions that do not lie between a<br>valid start and stop codon are called intergenic DNA and have other (possibly<br>unknown) functions. Computational biologists examine large DNA data files to<br>find patterns and important information, such as which regions are genes.<br>Sometimes they are interested in the percentages of mass accounted for by each<br>of the four nucleotide types. Often high percentages of Cytosine (C) and<br>Guanine (G) are indicators of important genetic data.<br>For more information, visit the Wikipedia page about DNA: <a target="_blank" rel="noopener" href="https://en.wikipedia.org/wiki/DNA"><br>http://en.wikipedia.org/wiki/DNA </a><br>In this assignment you read an input file containing named sequences of<br>nucleotides and produce information about them. For each nucleotide sequence,<br>your program counts the occurrences of each of the four nucleotides (A, C, G,<br>and T). The program also computes the mass percentage occupied by each<br>nucleotide type, rounded to one digit past the decimal point. Next the program<br>reports the codons (trios of nucleotides) present in each sequence and<br>predicts whether or not the sequence is a protein-coding gene. For us, a<br>protein-coding gene is a string that matches all of the following<br>constraints*:</p>
<ul>
<li>begins with a valid start codon (ATG) </li>
<li>ends with a valid stop codon (one of the following: TAA, TAG, or TGA) </li>
<li>contains at least 5 total codons (including its initial start codon and final stop codon) </li>
<li>Cytosine (C) and Guanine (G) combined account for at least 30% of its total mass<br>(*These are approximations for our assignment, not exact constraints used in<br>computational biology to identify proteins.)<br>The DNA input data consists of line pairs. The first line has the name of the<br>nucleotide sequence, and the second is the nucleotide sequence itself. Each<br>character in a sequence of nucleotides will be A, C, G, T, or a dash<br>character, “-“. The nucleotides in the input can be either upper or lowercase.<br>Input file dna.txt (partial):<br>cure for cancer protein<br>ATGCCACTATGGTAG<br>captain picard hair growth protein<br>ATgCCAACATGgATGCCcGATAtGGATTgA<br>bogus protein<br>CCATt-AATgATCa-CAGTt<br>The dash “-“ characters represent “junk” or “garbage” regions in the sequence.<br>For most of the program they should be ignored in your computations, though<br>they do contribute to the total mass of the sequence as described later.</li>
</ul>
<h2 id="Program-Behavior"><a href="#Program-Behavior" class="headerlink" title="Program Behavior"></a>Program Behavior</h2><p>Your program begins with an introduction and prompts for input and output file<br>names. You may assume the user will type the name of an existing input file<br>that is in the proper format. Your program reads the input file to process its<br>nucleotide sequences and outputs the results into the given output file.<br>Notice the nucleotide sequence is output in uppercase, and that the nucleotide<br>counts and mass percentages are shown in A, C, G, T order. A given codon such<br>as GAT might occur more than once in the same sequence.<br>Log of execution (user input underlined):<br>    This program reports information about DNA nucleotide sequences that may encode proteins.<br>    Input file name? dna.txt<br>    Output file name? output.txt<br>Output file output.txt after above execution (partial):<br>    Region Name: cure for cancer protein<br>    Nucleotides: ATGCCACTATGGTAG<br>    Nuc. Counts: [4, 3, 4, 4]<br>    Total Mass%: [27.3, 16.8, 30.6, 25.3] of 1978.8<br>    Codons List: [ATG, CCA, CTA, TGG, TAG]<br>    Is Protein?: YES<br>    Region Name: captain picard hair growth protein<br>    Nucleotides: ATGCCAACATGGATGCCCGATATGGATTGA<br>    Nuc. Counts: [9, 6, 8, 7]<br>    Total Mass%: [30.7, 16.8, 30.5, 22.1] of 3967.5<br>    Codons List: [ATG, CCA, ACA, TGG, ATG, CCC, GAT, ATG, GAT, TGA]<br>    Is Protein?: YES<br>    Region Name: bogus protein<br>    Nucleotides: CCATT-AATGATCA-CAGTT<br>    Nuc. Counts: [6, 4, 2, 6]<br>    Total Mass%: [32.3, 17.7, 12.1, 29.9] of 2508.1<br>    Codons List: [CCA, TTA, ATG, ATC, ACA, GTT]<br>    Is Protein?: NO</p>
<h2 id="Implementation-Guidelines-Hints-and-Development-Strategy"><a href="#Implementation-Guidelines-Hints-and-Development-Strategy" class="headerlink" title="Implementation Guidelines, Hints, and Development Strategy"></a>Implementation Guidelines, Hints, and Development Strategy</h2><p>The main purpose of this assignment is to demonstrate your understanding of<br>arrays and array traversals with for loops. Therefore, you should use arrays<br>to store the various data for each sequence. In particular, your nucleotide<br>counts, mass percentages, and codons should all be stored using arrays.<br>Additionally you should use arrays and for loops to transform the data from<br>one form to another as follows:</p>
<ul>
<li>from the original nucleotide sequence string to nucleotide counts; </li>
<li>from nucleotide counts to mass percentages; and </li>
<li>from the original nucleotide sequence string to codon triplets.<br>These transformations are summarized by the following diagram using the “cure<br>for cancer” protein data:<br>Nucleotides:<br>Recall that you can print any array using the method Arrays.toString. For<br>example:<br>int[] numbers &#x3D; {10, 20, 30, 40};<br>System.out.println(“my data is “ + Arrays.toString(numbers)); &#x2F;&#x2F; my data is [10, 20, 30, 40]<br>—|—</li>
</ul>
<p>To compute mass percentages, use the following as the mass of each nucleotide<br>(grams&#x2F;mol). The dashes representing “junk” regions are excluded from many<br>parts of your computations, but they do contribute mass to the total.</p>
<ul>
<li>Adenine (A): 135.128 </li>
<li>Cytosine (C): 111.103 </li>
<li>Guanine (G): 151.128 </li>
<li>Thymine (T): 125.107 </li>
<li>Junk (-): 100.000<br>For example, the mass of the sequence ATGG-AC is (135.128 + 125.107 + 151.128</li>
</ul>
<ul>
<li>151.128 + 100.000 + 135.128 + 111.103) or 908.722. Of this, 270.256 (29.7%)<br>is from the two Adenines; 111.103 (12.2%) is from the Cytosine; 302.256<br>(33.3%) is from the two Guanines; 125.107 (13.8%) is from the Thymine; and<br>100.000 (11.0%) is from the “junk” dash.<br>We suggest that you start this program by writing the code to read the input<br>file. Try writing code to simply read each protein’s name and sequence of<br>nucleotides and print them. Read each line from the input file using Scanner’s<br>nextLine method. This will read an entire line of input and return it as a<br>String.<br>Next, write code to pass over a nucleotide sequence and count the number of<br>As, Cs, Gs, and Ts. You can use a String’s charAt method to get individual<br>characters. Put your counts into an array of size 4. To map between<br>nucleotides and array indexes, you may want to write a method that converts a<br>single character (i.e. A, C, T, G) into indices (i.e. 0 to 3).<br>Once you have the counts working correctly, you can convert your counts into a<br>new array of percentages of mass for each nucleotide using the preceding<br>nucleotide mass values. If you’ve written code to map between nucleotide<br>letters and array indexes, it may also help you to look up mass values in an<br>array such as the following: double[] masses &#x3D; {135.128, 111.103, 151.128,<br>125.107};<br>You may store your mass percentages already rounded to one digit past the<br>decimal or you can round when printing the mass percentages array using<br>printf. If you choose to store the percentages pre-rounded, use Math.round as<br>follows:<br>  double num &#x3D; 1.6666667;<br>  double rounded &#x3D; Math.round(num * 10.0) &#x2F; 10.0;<br>  System.out.print(“the answer is “ + rounded);  &#x2F;&#x2F; the answer is 1.7<br>—|—<br>Remember that the “junk” dashes do contribute mass to the total. For other<br>parts of your program you may want to remove dashes from the input; consider<br>using the replace method on the nucleotide string to eliminate these<br>characters.<br>After computing mass percentages, you must break apart the sequence into<br>codons and examine each codon. You may wish to review the methods of String<br>objects as presented in Chapters 3 and 4, such as substring, charAt, indexOf,<br>replace, toUpperCase, and toLowerCase.<br>We also suggest that you first get your program working correctly printing its<br>output to the console before you save the output to a file. Once you have your<br>program printing correct output to the console, save the output to a file by<br>using a PrintStream as described in Section 6.4 of the textbook.<br>You may assume that the input file exists, is readable, and contains valid<br>input. (In other words, you should not re-prompt for input or output file<br>names.) You may assume that each sequence’s number of nucleotides (without<br>dashes) will be a multiple of 3, although the nucleotides on a line might be<br>in either uppercase or lowercase or a combination. Your program should<br>overwrite any existing data in the output file (this is the default<br>PrintStream behavior).</li>
</ul>
<h2 id="Style-Guidelines"><a href="#Style-Guidelines" class="headerlink" title="Style Guidelines"></a>Style Guidelines</h2><p>For this assignment you are required to have the following four class<br>constants:</p>
<ul>
<li>one for the minimum number of codons a valid protein must have, as an integer (default of 5) </li>
<li>a second for the percentage of mass from C and G in order for a protein to be valid, as an integer (default of 30) </li>
<li>a third for the number of unique nucleotides (4, representing A, C, G, and T) </li>
<li>a fourth for the number of nucleotides per codon (3)<br>For full credit it should be possible to change the first two constant values<br>(minimum codons and minimum mass percentage) and cause your program to change<br>its behavior for evaluating protein validity. The other two constants won’t<br>ever be changed but are still useful to make your program more readable. Refer<br>to these constants in your code and do not refer to the bare number such as 4<br>or 3 directly. You may use additional constants if they make your code<br>clearer.<br>We will grade your method structure strictly on this assignment. Use at least<br>four nontrivial methods besides main. These methods should use parameters and<br>returns, including arrays, as appropriate. The methods should be<br>wellstructured and avoid redundancy. No one method should do too large a share<br>of the overall task. The textbook’s case study at the end of Chapter 7 is a<br>good example of a larger program with methods that pass arrays as parameters.<br>In particular, we require that you have the following particular method in<br>your program:</li>
<li>A method to print all file output for a given potential protein (nucleotides, counts, %, is it a protein, etc.)<br>In other words, all output to the file should be done through one method<br>called on each nucleotide sequence from the input. Your other methods should<br>do the computations to gather information to be passed to this output method.<br>Your main method should be a concise summary of the overall program. It is<br>okay for main to contain some code such as println statements. But main should<br>not perform too large a share of the overall work itself, such as examining<br>each character of an input line. Also avoid “chaining,” when many methods call<br>each other without ever returning to main.<br>We will also check strictly for redundancy on this assignment. If you have a<br>very similar piece of code that is repeated several times in your program,<br>eliminate the redundancy such as by creating a method, by using for loops over<br>the elements of arrays, and&#x2F;or by factoring if&#x2F;else code as described in<br>section 4.3 of the textbook.<br>Since arrays are a key component of this assignment, part of your grade comes<br>from using arrays properly. For example, you should reduce redundancy as<br>appropriate by using traversals over arrays (for loops over the array’s<br>elements). This is preferable to writing out a separate statement for each<br>array element (a statement for element [0], then another for [1], then for<br>[2], etc.). Also carefully consider how arrays should be passed as parameters<br>and&#x2F;or returned from methods as you are decomposing your program. Recall that<br>arrays use reference semantics when passed as parameters, meaning that an<br>array passed to a method can be modified by that method and the changes will<br>be seen by the caller.<br>You are limited to features in Chapters 1 through 7. Follow past style<br>guidelines such as indentation, names, variables, types, line lengths, and<br>comments (at the beginning of your program, on each method, and on complex<br>sections of code).</li>
</ul>
<h2 id="Additional-Input-Files-Optional"><a href="#Additional-Input-Files-Optional" class="headerlink" title="Additional Input Files (Optional)"></a>Additional Input Files (Optional)</h2><p>If you would like to generate additional input files to test your program, you<br>can create them from actual NCBI genetic data. The following web site has many<br>data files that contain complete genomes for viruses:<br><a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/Viruses/">ftp://ftp.ncbi.nlm.nih.gov/genomes/Viruses/</a><br>The site contains many directories with names of organisms. After entering a<br>directory, you can find and save a genome file (a file whose name ends with<br>.fna) and a protein table (a file whose name ends with .ptt). On the course<br>web site we will provide you with a program to convert these .fna and .ptt<br>files into input files suitable for your homework.</p>

                
            </div>
            <hr/>

            

    <div class="reprint" id="reprint-statement">
        
            <div class="reprint__author">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-user">
                        文章作者:
                    </i>
                </span>
                <span class="reprint-info">
                    <a href="/about" rel="external nofollow noreferrer">SafePoker</a>
                </span>
            </div>
            <div class="reprint__type">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-link">
                        文章链接:
                    </i>
                </span>
                <span class="reprint-info">
                    <a href="http://saferman.github.io/2023/02/12/Java%E4%BB%A3%E5%86%99%EF%BC%9ACSE142DNA/">http://saferman.github.io/2023/02/12/Java%E4%BB%A3%E5%86%99%EF%BC%9ACSE142DNA/</a>
                </span>
            </div>
            <div class="reprint__notice">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-copyright">
                        版权声明:
                    </i>
                </span>
                <span class="reprint-info">
                    本博客所有文章除特別声明外，均采用
                    <a href="https://creativecommons.org/licenses/by/4.0/deed.zh" rel="external nofollow noreferrer" target="_blank">CC BY 4.0</a>
                    许可协议。转载请注明来源
                    <a href="/about" target="_blank">SafePoker</a>
                    !
                </span>
            </div>
        
    </div>

    <script async defer>
      document.addEventListener("copy", function (e) {
        let toastHTML = '<span>复制成功，请遵循本文的转载规则</span><button class="btn-flat toast-action" onclick="navToReprintStatement()" style="font-size: smaller">查看</a>';
        M.toast({html: toastHTML})
      });

      function navToReprintStatement() {
        $("html, body").animate({scrollTop: $("#reprint-statement").offset().top - 80}, 800);
      }
    </script>



            <div class="tag_share" style="display: block;">
                <div class="post-meta__tag-list" style="display: inline-block;">
                    
                        <div class="article-tag">
                            
                                <a href="/tags/Java%E4%BB%A3%E5%86%99/">
                                    <span class="chip bg-color">Java代写</span>
                                </a>
                            
                        </div>
                    
                </div>
                <div class="post_share" style="zoom: 80%; width: fit-content; display: inline-block; float: right; margin: -0.15rem 0;">
                    <link rel="stylesheet" type="text/css" href="/libs/share/css/share.min.css">
<div id="article-share">

    

    

</div>

                </div>
            </div>
            
                <div id="reward">
    <a href="#rewardModal" class="reward-link modal-trigger btn-floating btn-medium waves-effect waves-light red">赏</a>

    <!-- Modal Structure -->
    <div id="rewardModal" class="modal">
        <div class="modal-content">
            <a class="close modal-close"><i class="fas fa-times"></i></a>
            <h4 class="reward-title">你的赏识是我前进的动力</h4>
            <div class="reward-content">
                <div class="reward-tabs">
                    <ul class="tabs row">
                        <li class="tab col s12 wechat-tab waves-effect waves-light"><a href="#wechat">微 信</a></li>
                    </ul>
                    <div id="wechat">
                        <img src="/medias/newqrcode.jpg" class="reward-img" alt="微信打赏二维码">
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>

<script>
    $(function () {
        $('.tabs').tabs();
    });
</script>

            
        </div>
    </div>

    

    

    

    

    

    

    

    

    

<article id="prenext-posts" class="prev-next articles">
    <div class="row article-row">
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge left-badge text-color">
                <i class="fas fa-chevron-left"></i>&nbsp;上一篇</div>
            <div class="card">
                <a href="/2023/02/12/C%E4%BB%A3%E5%86%99%EF%BC%9ASENG265MTFandRLE/">
                    <div class="card-image">
                        
                        
                        <img src="/medias/featureimages/18.jpg" class="responsive-img" alt="C代写：SENG265MTFandRLE">
                        
                        <span class="card-title">C代写：SENG265MTFandRLE</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            
                        
                    </div>
                    <div class="publish-info">
                        <span class="publish-date">
                            <i class="far fa-clock fa-fw icon-date"></i>2023-02-12
                        </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-user fa-fw"></i>
                            SafePoker
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/C%E4%BB%A3%E5%86%99/">
                        <span class="chip bg-color">C代写</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge right-badge text-color">
                下一篇&nbsp;<i class="fas fa-chevron-right"></i>
            </div>
            <div class="card">
                <a href="/2023/02/12/Python%E4%BB%A3%E5%86%99%EF%BC%9ACSE6242ArgoLiteSQLite/">
                    <div class="card-image">
                        
                        
                        <img src="/medias/featureimages/10.jpg" class="responsive-img" alt="Python代写：CSE6242ArgoLiteSQLite">
                        
                        <span class="card-title">Python代写：CSE6242ArgoLiteSQLite</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            
                        
                    </div>
                    <div class="publish-info">
                            <span class="publish-date">
                                <i class="far fa-clock fa-fw icon-date"></i>2023-02-12
                            </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-user fa-fw"></i>
                            SafePoker
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/Python%E4%BB%A3%E5%86%99/">
                        <span class="chip bg-color">Python代写</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
    </div>
</article>

</div>



<!-- 代码块功能依赖 -->
<script type="text/javascript" src="/libs/codeBlock/codeBlockFuction.js"></script>



<!-- 代码语言 -->

<script type="text/javascript" src="/libs/codeBlock/codeLang.js"></script>


<!-- 代码块复制 -->

<script type="text/javascript" src="/libs/codeBlock/codeCopy.js"></script>


<!-- 代码块收缩 -->

<script type="text/javascript" src="/libs/codeBlock/codeShrink.js"></script>



    </div>
    <div id="toc-aside" class="expanded col l3 hide-on-med-and-down">
        <div class="toc-widget card" style="background-color: white;">
            <div class="toc-title"><i class="far fa-list-alt"></i>&nbsp;&nbsp;目录</div>
            <div id="toc-content"></div>
        </div>
    </div>
</div>

<!-- TOC 悬浮按钮. -->

<div id="floating-toc-btn" class="hide-on-med-and-down">
    <a class="btn-floating btn-large bg-color">
        <i class="fas fa-list-ul"></i>
    </a>
</div>


<script src="/libs/tocbot/tocbot.min.js"></script>
<script>
    $(function () {
        tocbot.init({
            tocSelector: '#toc-content',
            contentSelector: '#articleContent',
            headingsOffset: -($(window).height() * 0.4 - 45),
            collapseDepth: Number('0'),
            headingSelector: 'h2, h3, h4'
        });

        // Set scroll toc fixed.
        let tocHeight = parseInt($(window).height() * 0.4 - 64);
        let $tocWidget = $('.toc-widget');
        $(window).scroll(function () {
            let scroll = $(window).scrollTop();
            /* add post toc fixed. */
            if (scroll > tocHeight) {
                $tocWidget.addClass('toc-fixed');
            } else {
                $tocWidget.removeClass('toc-fixed');
            }
        });

        
        /* 修复文章卡片 div 的宽度. */
        let fixPostCardWidth = function (srcId, targetId) {
            let srcDiv = $('#' + srcId);
            if (srcDiv.length === 0) {
                return;
            }

            let w = srcDiv.width();
            if (w >= 450) {
                w = w + 21;
            } else if (w >= 350 && w < 450) {
                w = w + 18;
            } else if (w >= 300 && w < 350) {
                w = w + 16;
            } else {
                w = w + 14;
            }
            $('#' + targetId).width(w);
        };

        // 切换TOC目录展开收缩的相关操作.
        const expandedClass = 'expanded';
        let $tocAside = $('#toc-aside');
        let $mainContent = $('#main-content');
        $('#floating-toc-btn .btn-floating').click(function () {
            if ($tocAside.hasClass(expandedClass)) {
                $tocAside.removeClass(expandedClass).hide();
                $mainContent.removeClass('l9');
            } else {
                $tocAside.addClass(expandedClass).show();
                $mainContent.addClass('l9');
            }
            fixPostCardWidth('artDetail', 'prenext-posts');
        });
        
    });
</script>

    

</main>




    <footer class="page-footer bg-color">
    

    <div class="container row center-align"
         style="margin-bottom: 0px !important;">
        <div class="col s12 m8 l8 copy-right">
            Copyright&nbsp;&copy;
            
                <span id="year">2018-2024</span>
            
            <a href="/contact" target="_blank">SafePoker</a>
            
            <br>
            
            
            
                
            
            
                <span id="busuanzi_container_site_pv">
                &nbsp;|&nbsp;<i class="far fa-eye"></i>&nbsp;总访问量:&nbsp;
                    <span id="busuanzi_value_site_pv" class="white-color"></span>
            </span>
            
            
                <span id="busuanzi_container_site_uv">
                &nbsp;|&nbsp;<i class="fas fa-users"></i>&nbsp;总访问人数:&nbsp;
                    <span id="busuanzi_value_site_uv" class="white-color"></span>
            </span>
            
            <br>

            <!-- 运行天数提醒. -->
            
            <br>
            
        </div>
        <div class="col s12 m4 l4 social-link social-statis">


    <a href="mailto:hhh@gmail.com" class="tooltipped" target="_blank" data-tooltip="邮件联系我" data-position="top" data-delay="50">
        <i class="fas fa-envelope-open"></i>
    </a>













</div>
    </div>
</footer>

<div class="progress-bar"></div>


    <!-- 搜索遮罩框 -->
<div id="searchModal" class="modal">
    <div class="modal-content">
        <div class="search-header">
            <span class="title"><i class="fas fa-search"></i>&nbsp;&nbsp;搜索</span>
            <input type="search" id="searchInput" name="s" placeholder="请输入搜索的关键字"
                   class="search-input">
        </div>
        <div id="searchResult"></div>
    </div>
</div>

<script type="text/javascript">
$(function () {
    var searchFunc = function (path, search_id, content_id) {
        'use strict';
        $.ajax({
            url: path,
            dataType: "xml",
            success: function (xmlResponse) {
                // get the contents from search data
                var datas = $("entry", xmlResponse).map(function () {
                    return {
                        title: $("title", this).text(),
                        content: $("content", this).text(),
                        url: $("url", this).text()
                    };
                }).get();
                var $input = document.getElementById(search_id);
                var $resultContent = document.getElementById(content_id);
                $input.addEventListener('input', function () {
                    var str = '<ul class=\"search-result-list\">';
                    var keywords = this.value.trim().toLowerCase().split(/[\s\-]+/);
                    $resultContent.innerHTML = "";
                    if (this.value.trim().length <= 0) {
                        return;
                    }
                    // perform local searching
                    datas.forEach(function (data) {
                        var isMatch = true;
                        var data_title = data.title.trim().toLowerCase();
                        var data_content = data.content.trim().replace(/<[^>]+>/g, "").toLowerCase();
                        var data_url = data.url;
                        data_url = data_url.indexOf('/') === 0 ? data.url : '/' + data_url;
                        var index_title = -1;
                        var index_content = -1;
                        var first_occur = -1;
                        // only match artiles with not empty titles and contents
                        if (data_title !== '' && data_content !== '') {
                            keywords.forEach(function (keyword, i) {
                                index_title = data_title.indexOf(keyword);
                                index_content = data_content.indexOf(keyword);
                                if (index_title < 0 && index_content < 0) {
                                    isMatch = false;
                                } else {
                                    if (index_content < 0) {
                                        index_content = 0;
                                    }
                                    if (i === 0) {
                                        first_occur = index_content;
                                    }
                                }
                            });
                        }
                        // show search results
                        if (isMatch) {
                            str += "<li><a href='" + data_url + "' class='search-result-title'>" + data_title + "</a>";
                            var content = data.content.trim().replace(/<[^>]+>/g, "");
                            if (first_occur >= 0) {
                                // cut out 100 characters
                                var start = first_occur - 20;
                                var end = first_occur + 80;
                                if (start < 0) {
                                    start = 0;
                                }
                                if (start === 0) {
                                    end = 100;
                                }
                                if (end > content.length) {
                                    end = content.length;
                                }
                                var match_content = content.substr(start, end);
                                // highlight all keywords
                                keywords.forEach(function (keyword) {
                                    var regS = new RegExp(keyword, "gi");
                                    match_content = match_content.replace(regS, "<em class=\"search-keyword\">" + keyword + "</em>");
                                });

                                str += "<p class=\"search-result\">" + match_content + "...</p>"
                            }
                            str += "</li>";
                        }
                    });
                    str += "</ul>";
                    $resultContent.innerHTML = str;
                });
            }
        });
    };

    searchFunc('/search.xml', 'searchInput', 'searchResult');
});
</script>

    <!-- 白天和黑夜主题 -->
<div class="stars-con">
    <div id="stars"></div>
    <div id="stars2"></div>
    <div id="stars3"></div>  
</div>

<script>
    function switchNightMode() {
        $('<div class="Cuteen_DarkSky"><div class="Cuteen_DarkPlanet"></div></div>').appendTo($('body')),
        setTimeout(function () {
            $('body').hasClass('DarkMode') 
            ? ($('body').removeClass('DarkMode'), localStorage.setItem('isDark', '0'), $('#sum-moon-icon').removeClass("fa-sun").addClass('fa-moon')) 
            : ($('body').addClass('DarkMode'), localStorage.setItem('isDark', '1'), $('#sum-moon-icon').addClass("fa-sun").removeClass('fa-moon')),
            
            setTimeout(function () {
            $('.Cuteen_DarkSky').fadeOut(1e3, function () {
                $(this).remove()
            })
            }, 2e3)
        })
    }
</script>

    <!-- 回到顶部按钮 -->
<div id="backTop" class="top-scroll">
    <a class="btn-floating btn-large waves-effect waves-light" href="#!">
        <i class="fas fa-arrow-up"></i>
    </a>
</div>


    <script src="/libs/materialize/materialize.min.js"></script>
    <script src="/libs/masonry/masonry.pkgd.min.js"></script>
    <script src="/libs/aos/aos.js"></script>
    <script src="/libs/scrollprogress/scrollProgress.min.js"></script>
    <script src="/libs/lightGallery/js/lightgallery-all.min.js"></script>
    <script src="/js/matery.js"></script>

    

    
    
    

    <!-- 雪花特效 -->
    

    <!-- 鼠标星星特效 -->
    

     
        <script src="https://ssl.captcha.qq.com/TCaptcha.js"></script>
        <script src="/libs/others/TencentCaptcha.js"></script>
        <button id="TencentCaptcha" data-appid="xxxxxxxxxx" data-cbfn="callback" type="button" hidden></button>
    

    <!-- Baidu Analytics -->

    <!-- Baidu Push -->

<script>
    (function () {
        var bp = document.createElement('script');
        var curProtocol = window.location.protocol.split(':')[0];
        if (curProtocol === 'https') {
            bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';
        } else {
            bp.src = 'http://push.zhanzhang.baidu.com/push.js';
        }
        var s = document.getElementsByTagName("script")[0];
        s.parentNode.insertBefore(bp, s);
    })();
</script>

    
    <script src="/libs/others/clicklove.js" async="async"></script>
    
    
    <script async src="/libs/others/busuanzi.pure.mini.js"></script>
    

    

    

    
    

    

    

    
    <script src="/libs/instantpage/instantpage.js" type="module"></script>
    

</body>

</html>
